# 📊 数据加载函数使用手册

本 Notebook 展示 `data.py` 中各种数据加载函数的具体调用方法和返回结果。

## 一、 可用的数据加载函数

### 1. DataManager 类
- `DataManager.load_data()` - 核心数据加载方法
- `DataManager.get_stock_list()` - 获取股票列表  
- `DataManager.get_trading_dates()` - 获取交易日列表
- `DataManager.get_data_summary()` - 获取数据概览

### 2. 便捷函数
- `load_stock_data()` - 快速加载股票日线数据
- `load_index_data()` - 快速加载指数数据
- `load_financial_data()` - 快速加载财务数据

In [17]:
# 基础导入
import sys
import os
import pandas as pd
import numpy as np
from datetime import datetime, date
sys.path.append(os.path.dirname(os.path.abspath('.')))

from data import DataManager, load_stock_data, load_financial_data

## 二、 DataManager 类方法

In [18]:
# 创建 DataManager 实例
dm = DataManager()

print(" DataManager 初始化完成")
print(f" 数据目录: {dm.clean_data_path}")
print(f" 缓存状态: {'启用' if hasattr(dm, '_cache') else '禁用'}")

 DataManager 初始化完成
 数据目录: /Users/nachuanchen/Documents/Undergrad_Resources/投资学/Factor_strategy_platform/data_manager/clean_data
 缓存状态: 启用


### 1. load_data() 方法

In [19]:
# 方法1: 加载全部股票日线数据
print("=" * 50)
print("load_data('daily') - 加载全部股票数据")
print("=" * 50)

daily_data = dm.load_data('daily')

print(f"返回类型: {type(daily_data)}")
print(f"数据形状: {daily_data.shape}")
print(f"时间范围: {daily_data['trade_date'].min()} ~ {daily_data['trade_date'].max()}")
print(f"股票数量: {daily_data['ts_code'].nunique()}")
print(f"主要列: {list(daily_data.columns[:8])}")
print("\n前3行数据:")
print(daily_data.head(3)[['ts_code', 'trade_date', 'open', 'high', 'low', 'close']])

load_data('daily') - 加载全部股票数据
✅ 清洗后数据加载成功！
   数据类型: daily
   数据量: 10,216,839 条记录，13 列
返回类型: <class 'pandas.core.frame.DataFrame'>
数据形状: (10216839, 13)
时间范围: 2010-01-04 00:00:00 ~ 2025-09-30 00:00:00
股票数量: 4806
主要列: ['ts_code', 'trade_date', 'open', 'high', 'low', 'close', 'pre_close', 'change']

前3行数据:
     ts_code trade_date   open   high    low  close
0  000001.SZ 2010-01-04  24.52  24.58  23.68  23.71
1  000001.SZ 2010-01-05  23.75  23.90  22.75  23.30
2  000001.SZ 2010-01-06  23.25  23.25  22.72  22.90


In [ ]:
# 方法2: 时间过滤
print("=" * 50) 
print("📊 load_data('daily', start_date='2024-01-01', end_date='2024-03-31')")
print("=" * 50)

data_filtered = dm.load_data('daily', start_date='2024-01-01', end_date='2024-03-31')

print(f"过滤后数据形状: {data_filtered.shape}")
print(f"时间范围: {data_filtered['trade_date'].min()} ~ {data_filtered['trade_date'].max()}")
print(f"交易日数: {data_filtered['trade_date'].nunique()}")

 load_data('daily', start_date='2024-01-01', end_date='2024-03-31')
✅ 清洗后数据加载成功！
   数据类型: daily
   数据量: 10,216,839 条记录，13 列
   过滤后: 258,244 条记录 (过滤了 9,958,595 条)
过滤后数据形状: (258244, 13)
时间范围: 2024-01-02 00:00:00 ~ 2024-03-29 00:00:00
交易日数: 58


In [21]:
# 方法3: 股票代码过滤
print("\n" + "=" * 50)
print(" load_data('daily', stock_codes=['000001.SZ', '600000.SH'])")
print("=" * 50)
stocks = ['000001.SZ', '600000.SH']
data_stocks = dm.load_data('daily', stock_codes=stocks, start_date='2024-01-01')
print(f"指定股票数据形状: {data_stocks.shape}")
print(f"股票列表: {list(data_stocks['ts_code'].unique())}")
print("\n每只股票最新数据:")
print(data_stocks.groupby('ts_code').tail(1)[['ts_code', 'trade_date', 'close', 'pct_chg']])


 load_data('daily', stock_codes=['000001.SZ', '600000.SH'])
✅ 清洗后数据加载成功！
   数据类型: daily
   数据量: 10,216,839 条记录，13 列
   过滤后: 850 条记录 (过滤了 10,215,989 条)
指定股票数据形状: (850, 13)
股票列表: ['000001.SZ', '600000.SH']

每只股票最新数据:
           ts_code trade_date  close  pct_chg
3755     000001.SZ 2025-09-30  11.34  -0.2639
5438837  600000.SH 2025-09-30  11.90  -1.3267


In [26]:
# 方法3: 股票代码过滤
print("\n" + "=" * 50)
print("📊 load_data('daily', stock_codes=['000001.SZ', '600000.SH'])")
print("=" * 50)
stocks = ['000001.SZ', '600000.SH']
data_stocks = dm.load_data('daily', stock_codes=stocks, start_date='2024-01-01')
print(f"指定股票数据形状: {data_stocks.shape}")
print(f"股票列表: {list(data_stocks['ts_code'].unique())}")
print("\n每只股票最新数据:")
print(data_stocks.groupby('ts_code').tail(1)[['ts_code', 'trade_date', 'close', 'pct_chg']])


📊 load_data('daily', stock_codes=['000001.SZ', '600000.SH'])
✅ 从缓存加载 daily 数据
指定股票数据形状: (850, 13)
股票列表: ['000001.SZ', '600000.SH']

每只股票最新数据:
           ts_code trade_date  close  pct_chg
3755     000001.SZ 2025-09-30  11.34  -0.2639
5438837  600000.SH 2025-09-30  11.90  -1.3267


### 2. 加载指数数据

In [27]:
# 方法4: 加载指数数据
print("=" * 50)
print("📊 load_data('index')")
print("=" * 50)
index_data = dm.load_data('index', start_date='2024-01-01')
print(f"指数数据形状: {index_data.shape}")
print(f"时间范围: {index_data['trade_date'].min()} ~ {index_data['trade_date'].max()}")
print(f"指数数量: {index_data['ts_code'].nunique()}")

print(f"\n可用指数:")
indices = index_data['ts_code'].unique()
for i, idx in enumerate(indices[:5]):
    print(f"  {i+1}. {idx}")
if len(indices) > 5:
    print(f"  ... 还有 {len(indices)-5} 个指数")
    
print(f"\n指数最新数据:")
print(index_data.groupby('ts_code').tail(1)[['ts_code', 'trade_date', 'close', 'pct_chg']].head())

📊 load_data('index')
✅ 从缓存加载 index 数据
指数数据形状: (2550, 11)
时间范围: 2024-01-02 00:00:00 ~ 2025-09-30 00:00:00
指数数量: 6

可用指数:
  1. 000001.SH
  2. 000016.SH
  3. 000300.SH
  4. 000852.SH
  5. 000905.SH
  ... 还有 1 个指数

指数最新数据:
         ts_code trade_date      close  pct_chg
3825   000001.SH 2025-09-30  3882.7774   0.5242
7651   000016.SH 2025-09-30  2988.9386   0.5347
11477  000300.SH 2025-09-30  4640.6935   0.4468
15303  000852.SH 2025-09-30  7574.9631   1.0287
19129  000905.SH 2025-09-30  7412.3684   0.8409


### 3. 加载财务数据

In [25]:
# 方法5: 加载资产负债表数据
print("=" * 50)
print("📊 load_data('balancesheet')")
print("=" * 50)
balance_data = dm.load_data('balancesheet', start_date='2022-01-01')
print(f"资产负债表数据形状: {balance_data.shape}")
print(f"报告期数: {balance_data['ann_date'].nunique()}")
print(f"股票数量: {balance_data['ts_code'].nunique()}")
print(f"关键字段: {[col for col in balance_data.columns if 'total' in col.lower()][:3]}")

# 方法6: 加载利润表数据
print("\n" + "=" * 50)
print("📊 load_data('income')")
print("=" * 50)
income_data = dm.load_data('income', stock_codes=['000001.SZ', '000002.SZ'], start_date='2022-01-01')
print(f"利润表数据形状: {income_data.shape}")
print(f"包含营业收入字段: {'revenue' in income_data.columns}")
print(f"包含净利润字段: {'n_income' in income_data.columns}")

# 方法7: 加载现金流数据
print("\n" + "=" * 50)
print("📊 load_data('cashflow')")
print("=" * 50)
cashflow_data = dm.load_data('cashflow', start_date='2023-01-01')
print(f"现金流数据形状: {cashflow_data.shape}")
print(f"包含经营现金流字段: {'n_cashflow_act' in cashflow_data.columns}")

print(f"\n财务数据样本 (资产负债表前3行):")
key_cols = ['ts_code', 'ann_date', 'total_assets', 'total_liab']
available_cols = [col for col in key_cols if col in balance_data.columns]
print(balance_data[available_cols].head(3))

📊 load_data('balancesheet')
✅ 清洗后数据加载成功！
   数据类型: balancesheet
   数据量: 356,393 条记录，152 列
   过滤后: 89,473 条记录 (过滤了 266,920 条)
资产负债表数据形状: (89473, 152)
报告期数: 767
股票数量: 5432
关键字段: ['total_share', 'total_cur_assets', 'total_nca']

📊 load_data('income')
✅ 清洗后数据加载成功！
   数据类型: balancesheet
   数据量: 356,393 条记录，152 列
   过滤后: 89,473 条记录 (过滤了 266,920 条)
资产负债表数据形状: (89473, 152)
报告期数: 767
股票数量: 5432
关键字段: ['total_share', 'total_cur_assets', 'total_nca']

📊 load_data('income')
✅ 清洗后数据加载成功！
   数据类型: income
   数据量: 362,154 条记录，85 列
   过滤后: 33 条记录 (过滤了 362,121 条)
利润表数据形状: (33, 85)
包含营业收入字段: True
包含净利润字段: True

📊 load_data('cashflow')
✅ 清洗后数据加载成功！
   数据类型: cashflow
   数据量: 338,328 条记录，97 列
   过滤后: 83,016 条记录 (过滤了 255,312 条)
现金流数据形状: (83016, 97)
包含经营现金流字段: True

财务数据样本 (资产负债表前3行):
     ts_code   ann_date  total_assets    total_liab
0  000001.SZ 2025-08-23  1.206000e+10  5.058336e+09
1  000001.SZ 2025-08-23  1.206000e+10  5.058336e+09
2  000001.SZ 2025-04-19  1.206000e+10  5.058336e+09
✅ 清洗后数据加载成功！
   数据类型:

### 4. DataManager 辅助方法

In [28]:
# 方法8: 获取股票列表
print("=" * 50)
print("📊 dm.get_stock_list()")
print("=" * 50)
stock_list = dm.get_stock_list(exclude_st=True)
print(f"返回类型: {type(stock_list)}")
print(f"股票总数: {len(stock_list)}")
print(f"前10只股票: {stock_list[:10]}")

# 方法9: 获取交易日列表  
print("\n" + "=" * 50)
print("📊 dm.get_trading_dates()")
print("=" * 50)
trading_dates = dm.get_trading_dates(start_date='2024-01-01', end_date='2024-01-31')
print(f"返回类型: {type(trading_dates)}")
print(f"2024年1月交易日数: {len(trading_dates)}")
print(f"前5个交易日: {trading_dates[:5]}")

# 方法10: 获取数据概览
print("\n" + "=" * 50)
print("📊 dm.get_data_summary()")
print("=" * 50)
summary = dm.get_data_summary()
print(f"返回类型: {type(summary)}")
print(f"数据类型: {list(summary.keys())}")
for data_type, info in summary.items():
    if 'error' not in info:
        print(f"{data_type}: {info['rows']:,} 行, {info['columns']} 列")

📊 dm.get_stock_list()
返回类型: <class 'list'>
股票总数: 4806
前10只股票: ['000001.SZ', '000002.SZ', '000006.SZ', '000007.SZ', '000008.SZ', '000009.SZ', '000010.SZ', '000011.SZ', '000012.SZ', '000014.SZ']

📊 dm.get_trading_dates()
返回类型: <class 'list'>
股票总数: 4806
前10只股票: ['000001.SZ', '000002.SZ', '000006.SZ', '000007.SZ', '000008.SZ', '000009.SZ', '000010.SZ', '000011.SZ', '000012.SZ', '000014.SZ']

📊 dm.get_trading_dates()
返回类型: <class 'list'>
2024年1月交易日数: 22
前5个交易日: ['2024-01-02', '2024-01-03', '2024-01-04', '2024-01-05', '2024-01-08']

📊 dm.get_data_summary()
返回类型: <class 'list'>
2024年1月交易日数: 22
前5个交易日: ['2024-01-02', '2024-01-03', '2024-01-04', '2024-01-05', '2024-01-08']

📊 dm.get_data_summary()
返回类型: <class 'dict'>
数据类型: ['daily', 'cashflow', 'balancesheet', 'income', 'index']
daily: 10,216,839 行, 13 列
cashflow: 338,328 行, 97 列
balancesheet: 356,393 行, 152 列
income: 362,154 行, 85 列
index: 22,956 行, 11 列
返回类型: <class 'dict'>
数据类型: ['daily', 'cashflow', 'balancesheet', 'income', 'index']
daily

## 三、便捷函数

不需要创建 DataManager 实例，直接调用的快捷函数。

### 1. 便捷函数1: load_stock_data()

In [29]:
# 便捷函数1: load_stock_data()
print("=" * 50)
print("📊 load_stock_data()")
print("=" * 50)

# 加载指定时间范围股票数据
from data import load_stock_data, load_index_data, load_financial_data

result1 = load_stock_data(start_date='2024-01-01', end_date='2024-01-31')
print(f"load_stock_data(时间范围): {result1.shape}")

# 加载指定股票
result2 = load_stock_data(stock_codes=['000001.SZ', '600000.SH'], start_date='2024-01-01')
print(f"load_stock_data(指定股票): {result2.shape}")
print(f"股票列表: {list(result2['ts_code'].unique())}")

# 便捷函数2: load_index_data()
print("\n" + "=" * 50)
print("📊 load_index_data()")
print("=" * 50)
index_result = load_index_data(start_date='2024-01-01', end_date='2024-01-31')
print(f"load_index_data(): {index_result.shape}")
print(f"指数数量: {index_result['ts_code'].nunique()}")

# 便捷函数3: load_financial_data()
print("\n" + "=" * 50)
print("📊 load_financial_data()")
print("=" * 50)

# 加载资产负债表
balance_result = load_financial_data('balancesheet', end_date='2024-12-31')
print(f"资产负债表: {balance_result.shape}")

# 加载利润表（指定股票）
income_result = load_financial_data('income', stock_codes=['000001.SZ'], end_date='2024-12-31')
print(f"指定股票利润表: {income_result.shape}")

# 加载现金流
cashflow_result = load_financial_data('cashflow', end_date='2024-12-31')
print(f"现金流表: {cashflow_result.shape}")

📊 load_stock_data()
✅ 清洗后数据加载成功！
   数据类型: daily
   数据量: 10,216,839 条记录，13 列
✅ 清洗后数据加载成功！
   数据类型: daily
   数据量: 10,216,839 条记录，13 列
   过滤后: 97,444 条记录 (过滤了 10,119,395 条)
load_stock_data(时间范围): (97444, 13)
   过滤后: 97,444 条记录 (过滤了 10,119,395 条)
load_stock_data(时间范围): (97444, 13)
✅ 清洗后数据加载成功！
   数据类型: daily
   数据量: 10,216,839 条记录，13 列
✅ 清洗后数据加载成功！
   数据类型: daily
   数据量: 10,216,839 条记录，13 列
   过滤后: 850 条记录 (过滤了 10,215,989 条)
load_stock_data(指定股票): (850, 13)
股票列表: ['000001.SZ', '600000.SH']

📊 load_index_data()
✅ 清洗后数据加载成功！
   数据类型: index
   数据量: 22,956 条记录，11 列
   过滤后: 132 条记录 (过滤了 22,824 条)
load_index_data(): (132, 11)
指数数量: 6

📊 load_financial_data()
✅ 清洗后数据加载成功！
   数据类型: balancesheet
   数据量: 356,393 条记录，152 列
   过滤后: 344,721 条记录 (过滤了 11,672 条)
   过滤后: 850 条记录 (过滤了 10,215,989 条)
load_stock_data(指定股票): (850, 13)
股票列表: ['000001.SZ', '600000.SH']

📊 load_index_data()
✅ 清洗后数据加载成功！
   数据类型: index
   数据量: 22,956 条记录，11 列
   过滤后: 132 条记录 (过滤了 22,824 条)
load_index_data(): (132, 11)
指数数量: 6

📊 lo

### 2.  数据类型参数说明

### data_type 支持的值:
- `'daily'` - 股票日线数据（开高低收、成交量等）
- `'index'` - 指数数据
- `'balancesheet'` - 资产负债表数据  
- `'income'` - 利润表数据
- `'cashflow'` - 现金流量表数据

### 常用参数组合示例:

In [ ]:
# 常用参数组合示例
print("🔧 常用参数组合示例")
print("=" * 40)

# 示例1: 获取最近一年的股票数据
example1 = dm.load_data('daily', start_date='2023-01-01', end_date='2023-12-31')
print(f"1. 年度数据: dm.load_data('daily', start_date='2023-01-01', end_date='2023-12-31')")
print(f"   结果: {example1.shape}")

# 示例2: 获取特定股票的季度数据
example2 = dm.load_data('daily', 
                       stock_codes=['000001.SZ', '000002.SZ'], 
                       start_date='2024-01-01', 
                       end_date='2024-03-31')
print(f"\n2. 季度特定股票: stock_codes=['000001.SZ', '000002.SZ'], start_date='2024-01-01'")
print(f"   结果: {example2.shape}")

# 示例3: 获取财务数据（最近2年）
example3 = dm.load_data('balancesheet', start_date='2022-01-01')
print(f"\n3. 最近财务数据: dm.load_data('balancesheet', start_date='2022-01-01')")
print(f"   结果: {example3.shape}")

# 示例4: 禁用缓存
example4 = dm.load_data('daily', 
                       stock_codes=['000001.SZ'], 
                       start_date='2024-01-01',
                       use_cache=False)
print(f"\n4. 禁用缓存: use_cache=False")
print(f"   结果: {example4.shape}")

print(f"\n📝 提示: 所有日期参数支持 'YYYY-MM-DD' 格式的字符串")

## 四、 函数对比总结

| 函数类型 | 函数名 | 用途 | 过滤功能 | 推荐使用场景 |
|---------|--------|------|----------|--------------|
| **DataManager类** | `load_data()` | 核心数据加载 | ✅ 时间、股票、缓存 | 复杂查询、批量操作 |
| | `get_stock_list()` | 获取股票列表 | ✅ 日期、ST股票 | 获取可用股票 |
| | `get_trading_dates()` | 获取交易日 | ✅ 时间范围 | 获取日期序列 |
| | `get_data_summary()` | 数据概览 | ❌ | 快速了解数据规模 |
| **便捷函数** | `load_stock_data()` | 股票数据 | ✅ 时间、股票 | 日常股票数据查询 |
| | `load_index_data()` | 指数数据 | ✅ 时间 | 基准指数分析 |
| | `load_financial_data()` | 财务数据 | ✅ 时间、股票 | 基本面分析 |
| **兼容函数** | `load_data_file()` | 通用加载 | ❌ | 向后兼容（不推荐） |

### 💡 使用建议:
- **新项目**: 优先使用 `DataManager` 类或便捷函数
- **快速查询**: 使用便捷函数 `load_stock_data()` 等
- **复杂操作**: 使用 `DataManager` 类的完整功能
- **性能要求**: 启用缓存 `use_cache=True`

## 五、 数据字段详细说明

本部分详细说明各个数据类型中每个列名的含义和数据内容，便于用户理解和使用数据。

---

### 📈 1. 股票日线数据 (daily)

股票日线数据包含每日的交易信息，是量化分析的基础数据源。

| 列名 | 数据类型 | 说明 | 示例值 |
|------|----------|------|--------|
| `ts_code` | string | 股票代码（Tushare代码格式） | "000001.SZ", "600000.SH" |
| `trade_date` | datetime | 交易日期 | 2024-01-02 |
| `open` | float | 开盘价（元） | 23.50 |
| `high` | float | 最高价（元） | 24.80 |
| `low` | float | 最低价（元） | 23.20 |
| `close` | float | 收盘价（元） | 24.30 |
| `pre_close` | float | 前收盘价（元） | 23.45 |
| `change` | float | 涨跌额（元） | 0.85 |
| `pct_chg` | float | 涨跌幅（%） | 3.62 |
| `vol` | float | 成交量（手） | 123456.0 |
| `amount` | float | 成交金额（千元） | 300000.0 |
| `turnover_rate` | float | 换手率（%） | 2.35 |
| `turnover_rate_f` | float | 换手率（自由流通股） | 2.87 |

**数据特点**：
- 时间范围：2010年至今
- 频率：每日
- 已排除：ST股票、新股（上市不满一年）
- 数据质量：已清洗，无缺失值

### 📊 2. 指数数据 (index)

指数数据提供市场基准和行业指数的历史表现数据。

| 列名 | 数据类型 | 说明 | 示例值 |
|------|----------|------|--------|
| `ts_code` | string | 指数代码 | "000001.SH", "399001.SZ" |
| `trade_date` | datetime | 交易日期 | 2024-01-02 |
| `open` | float | 开盘点数 | 3156.78 |
| `high` | float | 最高点数 | 3189.32 |
| `low` | float | 最低点数 | 3145.67 |
| `close` | float | 收盘点数 | 3178.45 |
| `pre_close` | float | 前收盘点数 | 3156.78 |
| `change` | float | 涨跌点数 | 21.67 |
| `pct_chg` | float | 涨跌幅（%） | 0.69 |
| `vol` | float | 成交量（手） | 45678912.0 |
| `amount` | float | 成交金额（千元） | 123456789.0 |

**常见指数代码**：
- `000001.SH` - 上证指数
- `399001.SZ` - 深证成指  
- `399006.SZ` - 创业板指
- `000016.SH` - 上证50
- `000300.SH` - 沪深300
- `000905.SH` - 中证500

### 💰 3. 资产负债表数据 (balancesheet)

资产负债表反映企业在特定时点的财务状况，包含资产、负债和股东权益信息。

#### 🔑 核心字段

| 列名 | 数据类型 | 说明 | 示例值 |
|------|----------|------|--------|
| `ts_code` | string | 股票代码 | "000001.SZ" |
| `ann_date` | datetime | 公告日期 | 2024-04-20 |
| `f_ann_date` | datetime | 实际公告日期 | 2024-04-20 |
| `end_date` | datetime | 报告期 | 2024-03-31 |
| `report_type` | string | 报告类型 | "1季报", "中报", "3季报", "年报" |
| `comp_type` | string | 公司类型 | "1" |
| `total_share` | float | 期末总股本 | 19405918198.0 |
| `cap_rese` | float | 资本公积金 | 23696364234.5 |
| `undistr_porfit` | float | 未分配利润 | 45123789456.2 |

#### 📊 资产项目

| 列名 | 说明 | 示例值 |
|------|------|--------|
| `total_assets` | 资产总计 | 1206000000000.0 |
| `total_cur_assets` | 流动资产合计 | 567890123456.7 |
| `money_cap` | 货币资金 | 123456789012.3 |
| `trad_asset` | 交易性金融资产 | 45678901234.5 |
| `notes_receiv` | 应收票据 | 12345678901.2 |
| `accounts_receiv` | 应收账款 | 67890123456.7 |
| `oth_receiv` | 其他应收款 | 23456789012.3 |
| `prepayment` | 预付款项 | 8901234567.8 |
| `div_receiv` | 应收股利 | 1234567890.1 |
| `int_receiv` | 应收利息 | 567890123.4 |
| `inventories` | 存货 | 34567890123.4 |
| `amor_exp` | 长期待摊费用 | 5678901234.5 |
| `nca_within_1y` | 一年内到期的非流动资产 | 8901234567.8 |
| `sett_rsrv` | 结算备付金 | 2345678901.2 |
| `loanto_oth_bank_fi` | 拆出资金 | 6789012345.6 |
| `premium_receiv` | 应收保费 | 1234567890.1 |
| `reinsur_receiv` | 应收分保账款 | 4567890123.4 |
| `reinsur_res_receiv` | 应收分保合同准备金 | 7890123456.7 |
| `pur_resale_fa` | 买入返售金融资产 | 2345678901.2 |
| `oth_cur_assets` | 其他流动资产 | 5678901234.5 |
| `total_nca` | 非流动资产合计 | 638109876543.3 |
| `fa_avail_for_sale` | 可供出售金融资产 | 8901234567.8 |
| `htm_invest` | 持有至到期投资 | 3456789012.3 |
| `lt_eqt_invest` | 长期股权投资 | 6789012345.6 |
| `invest_real_estate` | 投资性房地产 | 1234567890.1 |
| `time_deposits` | 定期存款 | 4567890123.4 |
| `oth_assets` | 其他资产 | 7890123456.7 |
| `r_and_d` | 研发支出 | 2345678901.2 |
| `goodwill` | 商誉 | 5678901234.5 |
| `lt_amor_exp` | 长期待摊费用 | 8901234567.8 |
| `defer_tax_assets` | 递延所得税资产 | 3456789012.3 |
| `decr_in_disbur` | 发放贷款及垫款 | 6789012345.6 |
| `oth_nca` | 其他非流动资产 | 1234567890.1 |

#### 💸 负债项目

| 列名 | 说明 | 示例值 |
|------|------|--------|
| `total_liab` | 负债合计 | 505833600000.0 |
| `total_cur_liab` | 流动负债合计 | 234567890123.4 |
| `st_borr` | 短期借款 | 45678901234.5 |
| `trad_liab` | 交易性金融负债 | 8901234567.8 |
| `notes_payable` | 应付票据 | 23456789012.3 |
| `acct_payable` | 应付账款 | 67890123456.7 |
| `adv_receipts` | 预收款项 | 12345678901.2 |
| `sold_for_repur_fa` | 卖出回购金融资产款 | 5678901234.5 |
| `comm_payable` | 应付手续费及佣金 | 1234567890.1 |
| `payroll_payable` | 应付职工薪酬 | 8901234567.8 |
| `taxes_payable` | 应交税费 | 3456789012.3 |
| `int_payable` | 应付利息 | 2345678901.2 |
| `div_payable` | 应付股利 | 6789012345.6 |
| `oth_payable` | 其他应付款 | 23456789012.3 |
| `acc_exp` | 预提费用 | 5678901234.5 |
| `deferred_inc` | 递延收益 | 1234567890.1 |
| `st_bonds_payable` | 应付短期债券 | 8901234567.8 |
| `payable_to_reinsurer` | 应付分保账款 | 3456789012.3 |
| `rsrv_insur_cont` | 保险合同准备金 | 6789012345.6 |
| `acting_trading_sec` | 代理买卖证券款 | 2345678901.2 |
| `acting_uw_sec` | 代理承销证券款 | 5678901234.5 |
| `non_cur_liab_due_1y` | 一年内到期的非流动负债 | 12345678901.2 |
| `oth_cur_liab` | 其他流动负债 | 7890123456.7 |
| `total_ncl` | 非流动负债合计 | 271265709876.6 |
| `lt_borr` | 长期借款 | 89012345678.9 |
| `bonds_payable` | 应付债券 | 34567890123.4 |
| `lt_payable` | 长期应付款 | 6789012345.6 |
| `specific_payables` | 专项应付款 | 1234567890.1 |
| `estimated_liab` | 预计负债 | 4567890123.4 |
| `defer_tax_liab` | 递延所得税负债 | 7890123456.7 |
| `defer_inc_non_cur_liab` | 递延收益-非流动负债 | 2345678901.2 |
| `oth_ncl` | 其他非流动负债 | 5678901234.5 |

#### 🏛️ 股东权益项目

| 列名 | 说明 | 示例值 |
|------|------|--------|
| `total_hldr_eqy_exc_min_int` | 股东权益合计(不含少数股东权益) | 700166400000.0 |
| `total_hldr_eqy_inc_min_int` | 股东权益合计(含少数股东权益) | 700166400000.0 |
| `total_equity` | 归属于母公司所有者权益合计 | 700166400000.0 |
| `cap_stk` | 实收资本(或股本) | 19405918198.0 |
| `cap_rsrv` | 资本公积 | 23696364234.5 |
| `surplus_rsrv` | 盈余公积 | 45678901234.5 |
| `ordin_risk_rsrv` | 一般风险准备 | 8901234567.8 |
| `retained_earnings` | 未分配利润 | 45123789456.2 |
| `tsy_stk` | 库存股 | -1234567890.1 |
| `oth_eqt_tools` | 其他权益工具 | 2345678901.2 |
| `oth_eqt_tools_p_shr` | 其他权益工具:优先股 | 567890123.4 |
| `minority_int` | 少数股东权益 | 23456789012.3 |

### 📈 4. 利润表数据 (income)

利润表反映企业在一定期间的经营成果，包含收入、成本、费用和利润信息。

#### 🔑 核心字段

| 列名 | 数据类型 | 说明 | 示例值 |
|------|----------|------|--------|
| `ts_code` | string | 股票代码 | "000001.SZ" |
| `ann_date` | datetime | 公告日期 | 2024-04-20 |
| `f_ann_date` | datetime | 实际公告日期 | 2024-04-20 |
| `end_date` | datetime | 报告期 | 2024-03-31 |
| `report_type` | string | 报告类型 | "1季报", "中报", "3季报", "年报" |
| `comp_type` | string | 公司类型 | "1" |
| `basic_eps` | float | 基本每股收益 | 1.23 |
| `diluted_eps` | float | 稀释每股收益 | 1.21 |

#### 💰 收入项目

| 列名 | 说明 | 示例值 |
|------|------|--------|
| `revenue` | 营业总收入 | 123456789012.3 |
| `oper_rev` | 营业收入 | 123456789012.3 |
| `int_income` | 利息收入 | 45678901234.5 |
| `prem_earned` | 已赚保费 | 8901234567.8 |
| `comm_income` | 手续费及佣金收入 | 2345678901.2 |
| `n_commis_income` | 手续费及佣金净收入 | 1234567890.1 |
| `n_oth_income` | 其他经营净收益 | 567890123.4 |
| `n_oth_b_income` | 加:其他业务净收益 | 890123456.7 |
| `prem_income` | 保费业务收入 | 3456789012.3 |
| `out_prem` | 分出保费 | 678901234.5 |
| `une_prem_reser` | 提取未到期责任准备金 | 123456789.0 |
| `reins_income` | 分保费收入 | 456789012.3 |
| `n_sec_tb_income` | 证券买卖差价收入 | 789012345.6 |
| `n_sec_uw_income` | 证券承销业务净收益 | 234567890.1 |
| `n_asset_mg_income` | 受托客户资产管理业务净收益 | 567890123.4 |

#### 💸 成本费用项目

| 列名 | 说明 | 示例值 |
|------|------|--------|
| `total_cogs` | 营业总成本 | 87654321098.7 |
| `oper_cost` | 营业成本 | 76543210987.6 |
| `int_exp` | 利息支出 | 3456789012.3 |
| `comm_exp` | 手续费及佣金支出 | 1234567890.1 |
| `biz_tax_surchg` | 税金及附加 | 567890123.4 |
| `sell_exp` | 销售费用 | 2345678901.2 |
| `admin_exp` | 管理费用 | 4567890123.4 |
| `fin_exp` | 财务费用 | 890123456.7 |
| `assets_impair_loss` | 资产减值损失 | 234567890.1 |
| `prem_refund` | 退保金 | 678901234.5 |
| `compens_payout` | 赔付支出净额 | 1234567890.1 |
| `reser_insur_liab` | 提取保险责任准备金 | 456789012.3 |
| `div_payt` | 保单红利支出 | 123456789.0 |
| `reins_exp` | 分保费用 | 789012345.6 |
| `oper_exp` | 营业支出 | 2345678901.2 |
| `compens_payout_refu` | 摊回赔付支出 | 567890123.4 |
| `insur_reser_refu` | 摊回保险责任准备金 | 890123456.7 |
| `reins_cost_refund` | 摊回分保费用 | 234567890.1 |
| `other_bus_cost` | 其他业务成本 | 678901234.5 |

#### 📊 利润项目

| 列名 | 说明 | 示例值 |
|------|------|--------|
| `operate_profit` | 营业利润 | 35802468013.6 |
| `non_oper_income` | 营业外收入 | 1234567890.1 |
| `non_oper_exp` | 营业外支出 | 456789012.3 |
| `nca_disploss` | 非流动资产处置损失 | 123456789.0 |
| `total_profit` | 利润总额 | 36580246891.4 |
| `income_tax` | 所得税费用 | 5487037033.7 |
| `n_income` | 净利润 | 31093209857.7 |
| `n_income_attr_p` | 归属于母公司所有者的净利润 | 31093209857.7 |
| `minority_gain` | 少数股东损益 | 234567890.1 |
| `oth_compr_income` | 其他综合收益 | 567890123.4 |
| `t_compr_income` | 综合收益总额 | 31661099981.1 |
| `compr_inc_attr_p` | 归属于母公司所有者的综合收益总额 | 31661099981.1 |
| `compr_inc_attr_m_s` | 归属于少数股东的综合收益总额 | 345678901.2 |
| `ebit` | 息税前利润 | 39258358026.3 |
| `ebitda` | 息税折旧摊销前利润 | 42714146927.6 |
| `insurance_exp` | 保险业务支出 | 6789012345.6 |
| `undist_profit` | 年初未分配利润 | 45123789456.2 |
| `distable_profit` | 可分配利润 | 76217999313.9 |

### 💸 5. 现金流量表数据 (cashflow)

现金流量表反映企业在一定期间现金和现金等价物流入和流出的情况。

#### 🔑 核心字段

| 列名 | 数据类型 | 说明 | 示例值 |
|------|----------|------|--------|
| `ts_code` | string | 股票代码 | "000001.SZ" |
| `ann_date` | datetime | 公告日期 | 2024-04-20 |
| `f_ann_date` | datetime | 实际公告日期 | 2024-04-20 |
| `end_date` | datetime | 报告期 | 2024-03-31 |
| `report_type` | string | 报告类型 | "1季报", "中报", "3季报", "年报" |
| `comp_type` | string | 公司类型 | "1" |

#### 💰 经营活动现金流量

| 列名 | 说明 | 示例值 |
|------|------|--------|
| `n_cashflow_act` | 经营活动产生的现金流量净额 | 23456789012.3 |
| `cash_recp_sg_and_rs` | 销售商品、提供劳务收到的现金 | 134567890123.4 |
| `recp_tax_rends` | 收到的税费返还 | 1234567890.1 |
| `n_recp_oth_oper_act` | 收到其他与经营活动有关的现金 | 5678901234.5 |
| `c_inf_fr_oper_act` | 经营活动现金流入小计 | 141481359247.0 |
| `c_paid_goods_s` | 购买商品、接受劳务支付的现金 | 67890123456.7 |
| `cash_pay_beh_empl` | 支付给职工以及为职工支付的现金 | 8901234567.8 |
| `pay_all_typ_tax` | 支付的各项税费 | 12345678901.2 |
| `n_incr_insured_dep` | 客户存款和同业存放款项净增加额 | 23456789012.3 |
| `n_incr_loans_cb` | 向中央银行借款净增加额 | 3456789012.3 |
| `n_inc_borr_oth_fi` | 向其他金融机构拆入资金净增加额 | 4567890123.4 |
| `prem_fr_orig_contr` | 收到原保险合同保费取得的现金 | 5678901234.5 |
| `n_incr_insured_dep` | 保户储金净增加额 | 6789012345.6 |
| `n_recp_disp_plac_oth_fi` | 处置以公允价值计量且其变动计入当期损益的金融资产净增加额 | 7890123456.7 |
| `c_pay_oth_oper_act` | 支付其他与经营活动有关的现金 | 23456789012.3 |
| `c_outf_fr_oper_act` | 经营活动现金流出小计 | 118024570235.7 |

#### 🏗️ 投资活动现金流量

| 列名 | 说明 | 示例值 |
|------|------|--------|
| `n_cashflow_inv_act` | 投资活动产生的现金流量净额 | -12345678901.2 |
| `c_recp_return_invest` | 收回投资收到的现金 | 45678901234.5 |
| `c_recp_invest_income` | 取得投资收益收到的现金 | 6789012345.6 |
| `n_recp_disp_fiolta` | 处置固定资产、无形资产和其他长期资产收回的现金净额 | 234567890.1 |
| `n_recp_disp_sobu` | 处置子公司及其他营业单位收到的现金净额 | 345678901.2 |
| `stot_cash_recp_oth_inv_act` | 收到其他与投资活动有关的现金 | 1234567890.1 |
| `c_inf_fr_inv_act` | 投资活动现金流入小计 | 54282459261.5 |
| `c_paid_acq_const_fiolta` | 购建固定资产、无形资产和其他长期资产支付的现金 | 8901234567.8 |
| `c_paid_invest` | 投资支付的现金 | 56789012345.6 |
| `n_incr_pledge_loan` | 质押贷款净增加额 | 2345678901.2 |
| `n_incr_acq_sobu` | 取得子公司及其他营业单位支付的现金净额 | 567890123.4 |
| `c_pay_oth_inv_act` | 支付其他与投资活动有关的现金 | 1234567890.1 |
| `c_outf_fr_inv_act` | 投资活动现金流出小计 | 69838483828.1 |

#### 💼 筹资活动现金流量

| 列名 | 说明 | 示例值 |
|------|------|--------|
| `n_cash_flows_fnc_act` | 筹资活动产生的现金流量净额 | -8901234567.8 |
| `c_recp_cap_contrib` | 吸收投资收到的现金 | 2345678901.2 |
| `incl_cash_recp_saims` | 其中:子公司吸收少数股东投资收到的现金 | 234567890.1 |
| `c_recp_borrow` | 取得借款收到的现金 | 34567890123.4 |
| `proc_issue_bonds` | 发行债券收到的现金 | 5678901234.5 |
| `c_recp_oth_cash_fa` | 收到其他与筹资活动有关的现金 | 890123456.7 |
| `c_inf_fr_fnc_act` | 筹资活动现金流入小计 | 43462593715.9 |
| `c_prepay_amt_borr` | 偿还债务支付的现金 | 34567890123.4 |
| `c_pay_dist_dpcp_int_exp` | 分配股利、利润或偿付利息支付的现金 | 12345678901.2 |
| `incl_dvd_profit_paid_sc_ms` | 其中:子公司支付给少数股东的股利、利润 | 345678901.2 |
| `pay_cash_oth_fnc_act` | 支付其他与筹资活动有关的现金 | 4567890123.4 |
| `c_outf_fr_fnc_act` | 筹资活动现金流出小计 | 52363828283.7 |

#### 📊 现金净变动情况

| 列名 | 说明 | 示例值 |
|------|------|--------|
| `c_cash_equ_end_period` | 期末现金及现金等价物余额 | 123456789012.3 |
| `c_cash_equ_beg_period` | 期初现金及现金等价物余额 | 109876543210.9 |
| `n_incr_cash_cash_equ` | 现金及现金等价物净增加额 | 13580245801.4 |
| `im_n_incr_cash_equ` | 加:期初现金及现金等价物余额 | 109876543210.9 |

**数据特点**：
- 报告频率：季度报告
- 数据单位：人民币元
- 现金流分类：经营活动、投资活动、筹资活动
- 净额列示：部分项目按净额列示

## ⚠️ 数据使用注意事项

### 📅 时间说明
- **交易日期** (`trade_date`): 实际交易发生的日期
- **公告日期** (`ann_date`): 公司发布财务报告的日期  
- **报告期** (`end_date`): 财务数据对应的会计期间结束日期

### 💰 金额单位
- **股票价格**: 人民币元
- **成交金额**: 千元
- **财务数据**: 人民币元（大部分为亿元量级）

### 🔍 数据质量
- **缺失值处理**: 已使用前向填充方法处理
- **异常值**: 已进行基本的异常值检测和处理
- **ST股票**: 已从日线数据中移除
- **新股**: 已移除上市不满一年的股票

### 📊 使用建议
1. **因子计算**: 建议使用日线数据中的价格和成交量信息
2. **基本面分析**: 结合资产负债表、利润表、现金流量表数据
3. **时间对齐**: 注意财务数据的公告日期与交易日期的时间差
4. **数据更新**: 数据更新至2025年9月，建议定期检查数据时效性

### 🚀 快速上手
```python
# 基本用法示例
dm = DataManager()

# 获取最近一年的股票数据
stock_data = dm.load_data('daily', start_date='2024-01-01')

# 获取特定股票的财务数据
financial_data = dm.load_data('balancesheet', 
                             stock_codes=['000001.SZ'], 
                             start_date='2023-01-01')

# 使用便捷函数
recent_data = load_stock_data(start_date='2024-01-01')
```

---

📚 **更多信息请参考**: 
- [Tushare API文档](https://tushare.pro/document/2)
- [Wind数据字段说明](https://www.wind.com.cn/)
- [财务报表分析手册](https://www.example.com)